core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -80,28 +80,44 @@ class AdminManager(val config: KafkaConfig,
             && !arguments.replicasAssignments.isEmpty)
             throw new InvalidRequestException("Both numPartitions or replicationFactor and replicasAssignments were set. " +
               "Both cannot be used at the same time.")
-          else {
-            createTopicPolicy.foreach(_.validate(new RequestMetadata(topic, arguments.numPartitions,
-              arguments.replicationFactor, arguments.replicasAssignments, arguments.configs)))
-
-            if (!arguments.replicasAssignments.isEmpty) {
-              // Note: we don't check that replicaAssignment doesn't contain unknown brokers - unlike in add-partitions case,
-              // this follows the existing logic in TopicCommand
-              arguments.replicasAssignments.asScala.map { case (partitionId, replicas) =>
-                (partitionId.intValue, replicas.asScala.map(_.intValue))
-              }
-            } else {
-              AdminUtils.assignReplicasToBrokers(brokers, arguments.numPartitions, arguments.replicationFactor)
+          else if (!arguments.replicasAssignments.isEmpty) {
+            // Note: we don't check that replicaAssignment doesn't contain unknown brokers - unlike in add-partitions case,
```

Comment:
Removed the double negative. I am guessing it's this way because you may have a broker that is down temporarily, but I am not really sure.